### DISCUSSION TITLE
Will Action account-only events now be possible?

### DISCUSSION THREAD

**Discussion by Andre601** [upptime/upptime#649] [Q&A]
GitHub made a new Changelog post announcing that `workflow_dispatch` and `repository_dispatch` are now triggerable through the `GITHUB_TOKEN`. ([Blog post](https://github.blog/changelog/2022-09-08-github-actions-use-github_token-with-workflow_dispatch-and-repository_dispatch/))

My question now is, could upptime profit from this change? From my knowledge is it scheduling events and using dispatch triggers to create the other events on a set interval. This so far required the PAT to work properly. But with that change, if I understood it correctly, should the PAT now no longer be needed?

**Comment by AnandChowdhary** [comment]
Yes, I suggest we test using just the GITHUB_TOKEN and if everything works, we update the template and docs to recommend no longer using a PAT. :)

**Reply by Andre601** [reply]
How can I test this?

Also, through trying to work with GitHub's new Projects Beta have I discovered this [Docs page](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions) mentioning a setup, where you can use an Action ([tibdex/github-app-token](https://github.com/tibdex/github-app-token)) to use a GitHub App as a "user" to perform actions that would require a PAT.
At least for organisations would this be a better approach as the owner/member of the org doesn't need to give their own PAT or create a new account just for this, but could instead utilize a GitHub App they create.

This, obviously, is useless if the actions can now be run with a standard token instead, but if it doesn't work as expected should the setup perhaps have an option (Maybe using an `if` statement with env variables?) to support this alternative approach...

EDIT: Maybe have the Upptime-Bot be an actual bot and use that (^) approach for it? Seems a bit cleaner imo.

**Reply by Andre601** [reply]
@AnandChowdhary I quickly tested it by removing the PAT from the repository's secrets, but that doesn't work.
The Uptime CI fails, as the Checkout action receives a not-provided token (nothing), so there needs to be some work done to test this out.

Also, perhaps a look in that app-based thing I mentioned could be useful?

**Reply by AnandChowdhary** [reply]
Damn. Yeah the app idea makes a ton of sense, except that we'd have to maintain Upptime servers (or use a serverless API system) whereas right now it's all on GitHub without any maintenance required.

**Reply by Andre601** [reply]
> Damn. Yeah the app idea makes a ton of sense, except that we'd have to maintain Upptime servers (or use a serverless API system) whereas right now it's all on GitHub without any maintenance required.

Not really.
As mentioned can you use an action to generate a token for the app.
Only thing required would be the pem key for the app, which can be kept in a secret.

**Reply by Andre601** [reply]
To give a bit more context:
In an Org I'm part of did I create a GitHub App for it and use that one for sending canned responses for specific labels.

Using [`tibdex/github-app-token`](https://github.com/tibdex/github-app-token) I create a token for the App to use in the follow-up steps/jobs to use.

The workflow, in case you're curious: https://github.com/Readme-Workflows/recent-activity/blob/main/.github/workflows/comment.yml

So, what a user could do is create a GitHub App, save the PEM Key and App id as secrets and use those in upptime.
Upptime would then utilize github-app-token to generate the token and use that token in all of the action runs.

Also, in what way is the upptime-bot account used? I see commits from it in my repo, so it must do something outside my repo (Everything else doesn't make sense), so if stuff is done in one of upptime's repositories (which I assume) should it be possible to host the secrets there and use them.
Also, Org-wide secrets are afaik also possible...

**Reply by AnandChowdhary** [reply]
Hmm and how do you think we should deal with storing the private key? I don't think it makes sense to share it with all repos, right, so we still probably need a secured endpoint that returns a new app token?

**Reply by Andre601** [reply]
It depends on how the action(s) interact with the user.
